### PR TITLE
Add certificate chain reconstruction logic

### PR DESCRIPTION
**Pull request by deeglaze**
If a machine that produces an attestation report isn't "provisioned" with offline copies of its certificate chain, either the client or the verifier has to use AMD's Key Distribution Service to get the associated certificates. This PR adds logic to download the required certificates and construct an Attestation proto from just a Report or its raw bytes equivalent.

### CODE REVIEW

File: verify/kds/kds.go [hunk 1]
```diff
@@ -361,3 +364,27 @@ func ParsePlatformCertChain(pems []byte) ([]byte, []byte, error) {
 	}
 	return askBlock.Bytes, arkBlock.Bytes, nil
 }
+
+func platformBaseURL(name string) string {
```
Comment (kongoshuu) [inline verify/kds/kds.go:368]: Comment to describe the method?

File: verify/verify_test.go [hunk 2]
```diff
@@ -414,3 +417,17 @@ func TestOpenGetExtendedReportVerifyClose(t *testing.T) {
...
 	}
 }
+
+func TestRealAttestationVerification(t *testing.T) {
+	var nonce [64]byte
+	copy(nonce[:], []byte{1, 2, 3, 4, 5})
+	getter := &test.Getter{
+		Responses: map[string][]byte{
+			"https://kdsintf.amd.com/vcek/v1/Milan/cert_chain": milanBytes,
+			"https://kdsintf.amd.com/vcek/v1/Milan/3ac3fe21e13fb0990eb28a802e3fb6a29483a6b0753590c951bdd3b8e53786184ca39e359669a2b76a1936776b564ea464cdce40c05f63c9b610c5068b006b5d?blSPL=2&teeSPL=0&snpSPL=5&ucodeSPL=68": vcekBytes,
```
Comment (kongoshuu) [inline verify/verify_test.go:427]: Can you add some description for this line? It's not immediately obvious how this URL is generated

**Review by kongoshuu — APPROVED** [review]

File: verify/verify_test.go (hunk 2 above)
Comment (deeglaze) [reply verify/verify_test.go:428]: Done.

File: verify/kds/kds.go (hunk 1 above)
Comment (deeglaze) [reply verify/kds/kds.go:369]: Done, but the style guide doesn't require comments on unexported functions.

**Pull request merged**